contracts/src/v0.8/dev/ocr2dr/OCR2DRClient.sol

```diff
@@ -32,11 +34,18 @@ abstract contract OCR2DRClient is OCR2DRClientInterface {
   /**
    * @notice Sends OCR2DR request to the stored oracle address
    * @param req The initialized OCR2DR.Request
-   * @param subscriptionId The subscription ID
+   * @param billing The request's billing configuration
    * @return requestId The generated request ID
    */
-  function sendRequest(OCR2DR.Request memory req, uint256 subscriptionId) internal returns (bytes32) {
-    bytes32 requestId = s_oracle.sendRequest(subscriptionId, OCR2DR.encodeCBOR(req));
+  function sendRequest(OCR2DR.Request memory req, OCR2DRBillingInterface.RequestBilling calldata billing)
+    internal
+    returns (bytes32)
+  {
+    allowance = ERC20(billing.feeToken).allowance(address(this), address(s_oracle));
```

Comment:
Please help me understand what prevents user from specifying a shitty ERC20 feeToken? Should not the desired token to be set by Oracle?

Reply:
The Oracle contract extends the Billing abstract.
The Billing abstract contains a storage level config:
``` 
struct BillingConfig {
    uint32 gasOverhead;
    mapping(address => address) feeTokenPriceOracles; /* feeToken => oracle */
    mapping(address => uint32) requiredFeeByToken; /* feeToken => fee */
  }
```
That second field for `feeToken` contains a map for price for ERC20 token to CL price feed.
In a way this also functions as permissions for which ERC20 tokens are accepted.
If a `feeToken => oracle` is not added into the Oracle contract config, it will [revert](https://github.com/smartcontractkit/chainlink/pull/7814/files#diff-3ed2e725e380757b9506a577aba82c30c9c769c6784e979ad9644f5e7875e75bR51) - blocking the user from using a shitcoin.

Reply:
Makes me realize I should add a helper to the Oracle for `isFeeTokenSupported`